homeassistant/components/binary_sensor/rest.py

```diff
@@ -77,9 +77,17 @@ def is_on(self):
             return False
 
         if self._value_template is not None:
-            self.rest.data = template.render_with_possible_json_value(
+            self._state = template.render_with_possible_json_value(
                 self._hass, self._value_template, self.rest.data, False)
-        return bool(int(self.rest.data))
+        else:
```

Comment:
It should not be an else but always happen because it's a binary sensor and thus should be a boolean. Templates always return a string